pkg/pool-weighted/contracts/WeightedPool.sol

```diff
@@ -401,4 +403,51 @@ contract WeightedPool is BaseWeightedPool, WeightedPoolProtocolFees {
     {
         return super._isOwnerOnlyAction(actionId);
     }
+
+    function getSwapFeePercentage(bytes memory userData,
+     OperationType _operation) public view virtual override returns (uint256 _fee) {
```

Comment:
Alright, this doesn't seem to be a bad pattern. Assuming `isCustomFeeEnabled` is always true, then only those who can register the custom fee go through a different codepath when compared to the actual weighted pool.